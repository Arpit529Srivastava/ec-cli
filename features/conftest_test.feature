Feature: conftest test mode
  The ec test command should work as expected

  Scenario: success
    When ec command is run with "test --policy acceptance/examples/happy_day.rego acceptance/examples/empty_input.json -o json"
    Then the exit status should be 0
    Then the output should match the snapshot

  Scenario: a warning
    When ec command is run with "test --policy acceptance/examples/warn.rego acceptance/examples/empty_input.json -o json"
    Then the exit status should be 0
    Then the output should match the snapshot

  Scenario: a warning with fail-on-warn
    When ec command is run with "test --fail-on-warn -p acceptance/examples/warn.rego acceptance/examples/empty_input.json -o json"
    Then the exit status should be 1
    Then the output should match the snapshot

  Scenario: a deny
    When ec command is run with "test -p acceptance/examples/fail_with_data.rego --data acceptance/examples/rule_data_1.yaml acceptance/examples/empty_input.json -o json"
    Then the exit status should be 1
    Then the output should match the snapshot

  Scenario: a deny with no-fail
    When ec command is run with "test --no-fail -p acceptance/examples/fail_with_data.rego -d acceptance/examples/rule_data_1.yaml acceptance/examples/empty_input.json -o json"
    Then the exit status should be 0
    Then the output should match the snapshot

  Scenario: plain text deny
    When ec command is run with "test -p acceptance/examples/fail_with_data.rego --data acceptance/examples/rule_data_1.yaml acceptance/examples/empty_input.json --no-color"
    Then the exit status should be 1
    Then the output should match the snapshot
