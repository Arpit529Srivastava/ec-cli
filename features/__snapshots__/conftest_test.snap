
[success:stdout - 1]
[
  {
    "filename": "acceptance/examples/empty_input.json",
    "namespace": "main",
    "successes": 1
  }
]
---

[success:stderr - 1]

---
[a warning:stdout - 1]
[
  {
    "filename": "acceptance/examples/empty_input.json",
    "namespace": "main",
    "successes": 0,
    "warnings": [
      {
        "msg": "Has a warning"
      }
    ]
  }
]
---

[a warning:stderr - 1]

---
[a warning with fail-on-warn:stdout - 1]
[
  {
    "filename": "acceptance/examples/empty_input.json",
    "namespace": "main",
    "successes": 0,
    "warnings": [
      {
        "msg": "Has a warning"
      }
    ]
  }
]
---

[a warning with fail-on-warn:stderr - 1]

---
[a deny:stdout - 1]
[
  {
    "filename": "acceptance/examples/empty_input.json",
    "namespace": "main",
    "successes": 0,
    "failures": [
      {
        "msg": "Failure due to overripeness"
      }
    ]
  }
]
---

[a deny:stderr - 1]

---
[a deny with no-fail:stdout - 1]
[
  {
    "filename": "acceptance/examples/empty_input.json",
    "namespace": "main",
    "successes": 0,
    "failures": [
      {
        "msg": "Failure due to overripeness"
      }
    ]
  }
]
---

[a deny with no-fail:stderr - 1]

---
[plain text deny:stdout - 1]
FAIL - acceptance/examples/empty_input.json - main - Failure due to overripeness

1 test, 0 passed, 0 warnings, 1 failure, 0 exceptions

---

[plain text deny:stderr - 1]

---
